lib/rmg/CRmgTemplateZone.cpp

```diff
@@ -1297,27 +1304,19 @@ bool CRmgTemplateZone::placeMines ()
 	static const Res::ERes woodOre[] = {Res::ERes::WOOD, Res::ERes::ORE};
 	static const Res::ERes preciousResources[] = {Res::ERes::GEMS, Res::ERes::CRYSTAL, Res::ERes::MERCURY, Res::ERes::SULFUR};
 
-	std::array<TObjectTypeHandler, 7> factory =
-	{
-		VLC->objtypeh->getHandlerFor(Obj::MINE, 0),
-		VLC->objtypeh->getHandlerFor(Obj::MINE, 1),
-		VLC->objtypeh->getHandlerFor(Obj::MINE, 2),
-		VLC->objtypeh->getHandlerFor(Obj::MINE, 3),
-		VLC->objtypeh->getHandlerFor(Obj::MINE, 4),
-		VLC->objtypeh->getHandlerFor(Obj::MINE, 5),
-		VLC->objtypeh->getHandlerFor(Obj::MINE, 6)
-	};
+    std::vector<CGMine*> tempmines;
 
 	for (const auto & res : woodOre)
 	{
 		for (int i = 0; i < mines[res]; i++)
 		{
-			auto mine = (CGMine *) factory.at(static_cast<si32>(res))->create(ObjectTemplate());
+			auto mine = (CGMine*) VLC->objtypeh->getHandlerFor(Obj::MINE, res)->create(ObjectTemplate());
+            tempmines.emplace_back(mine);
 			mine->producedResource = res;
 			mine->tempOwner = PlayerColor::NEUTRAL;
 			mine->producedQuantity = mine->defaultResProduction();
```

Comment:
This section of code is repeated a few times. I suggest to introduce a factory method to create a mine by res type and set these default settings and afterwards we will only need to call addRequiredObject/addCloseObject

Reply:
Thanks for advise! I decided to change a bit this code but keep the idea to remove duplications